web/controllers/screeningController/index.ts

```diff
@@ -347,6 +374,67 @@ export async function updateCourse(req: Request, res: Response) {
     }
 }
 
+async function handleNewLectures(lectures: { subcourse: { id: number }, start: Date, duration: number, instructor: { id: number } }[], courseId: number) {
+    const entityManager = getManager();
+
+    for (let i=0; i<lectures.length; i++){
+        const course = await entityManager.findOne(Course, { id: courseId });
+        if (course == undefined) {
+            logger.warn(`No course found with ID ${courseId}`);
+            return 404;
+        }
+        const subcourse = await entityManager.findOne(Subcourse, { id: lectures[i].subcourse.id, course: course });
```

Comment:
This could possible be moved to the Course Model, like `const subcourse = (await course.getSubcourses())[0]`, to make it reusable

Reply:
The data modelling theoretically allows more than one subcourse with lectures though this possibility is not used at the moment. So finding the correct subcourse should always take into account the lecture here. 
However adding a function getSubcourse taking a lecture id to the course model would be valid, nevertheless this use case is in my opinion too fine-grained and specific for moving that function to the course model. I would leave it here.

Reply:
Ah, right, `Subcourse.forLecture(lecture)` would then be some way to move it to the Model. However I'm fine leaving it as is.